src/lightning/data/streaming/dataset_optimizer.py

```diff
@@ -798,3 +853,34 @@ def _upload_index(self, cache_dir: str, num_nodes: int, node_rank: Optional[int]
             merge_cache = Cache(cache_dir, chunk_bytes=1)
             merge_cache._merge_no_wait()
             self._upload_index(cache_dir, 1, None)
+
+    def _cleanup_cache(self) -> None:
+        cache_dir = _get_cache_dir(self.name)
+
+        # Cleanup the cache dir folder to avoid corrupted files from previous run to be there.
+        if os.path.exists(cache_dir):
+            rmtree(cache_dir)
+
+        os.makedirs(cache_dir, exist_ok=True)
+
+        cache_data_dir = _get_cache_data_dir(self.name)
+
+        # Cleanup the cache data folder to avoid corrupted files from previous run to be there.
+        if os.path.exists(cache_data_dir):
+            rmtree(cache_data_dir)
+
+        os.makedirs(cache_data_dir, exist_ok=True)
+
+    def _broadcast_object(self, obj: Any) -> Any:
+        """Enable to synchornize an object across machines using torch.distributed.collectives."""
+        num_nodes = _get_num_nodes()
+        if num_nodes == 1:
+            return obj
+
+        if not _distributed_is_initialized():
+            process_group_backend = "nccl" if is_cuda_available() else "gloo"
```

Comment:
`nccl` will require machines see each other via private IP otherwise it will hang, just calling it out

Reply:
Yes, the plugin takes care of that. So all good.